resource-server/src/main/java/com/okta/spring/example/ResourceServerExampleApplication.java

```diff
@@ -50,7 +51,7 @@ protected void configure(HttpSecurity http) throws Exception {
 
         @GetMapping("/api/userProfile")
         @PreAuthorize("hasAuthority('SCOPE_profile')")
-        public Map<String, Object> getUserDetails(JwtAuthenticationToken authentication) {
+        public <A extends AbstractOAuth2TokenAuthenticationToken<AbstractOAuth2Token>> Map<String, Object> getUserDetails(A authentication) {
```

Comment:
Add a comment about how this works with both `JwtAuthenticationToken` and OpaqueToken (i forget the exact class name)

Reply:
Added samples for both `JwtAuthenticationToken` and `BearerTokenAuthentication` cases